DDD.Functions/SubmitSessionFeedback.cs

```diff
@@ -0,0 +1,91 @@
+            SessionsConfig sessionsConfig
+            )
+        {
+            var feedback = await req.Content.ReadAsAsync<SessionFeedbackRequest>();
+
+            // Within feedback window
+            if (keyDates.Before(x => x.FeedbackAvailableFromDate) || keyDates.After(x => x.FeedbackAvailableToDate))
+            {
+                log.LogWarning("Attempt to access SubmitSessionFeedback endpoint outside of allowed window of {start} -> {end}.", keyDates.FeedbackAvailableFromDate, keyDates.FeedbackAvailableToDate);
+                return new StatusCodeResult((int)HttpStatusCode.NotFound);
+            }
+
+            // Valid feedback 
+            if ((feedback.Rating < 1 || feedback.Rating > 5) && string.IsNullOrWhiteSpace(feedback.Likes) && string.IsNullOrWhiteSpace(feedback.ImprovementIdeas))
+            {
+                log.LogWarning("Attempt to access SubmitSessionFeedback endpoint with invalid feedback details from {DeviceId}", feedback.DeviceId);
+                return new StatusCodeResult((int)HttpStatusCode.BadRequest);
+            }
+
+            // valid Session existed 
+            var (sessionsRepo, presentersRepo) = await sessionsConfig.GetRepositoryAsync();
+            var allSessions = await sessionsRepo.GetAllAsync(conference.ConferenceInstance);
+            var allPresenters = await presentersRepo.GetAllAsync(conference.ConferenceInstance);
+
+            var sessionRow = allSessions.First(s => s != null && feedback.SessionId == s.Id.ToString());
+            if (sessionRow == null)
+            {
+                log.LogWarning("Attempt to submit to SubmitSessionFeedback endpoint with invalid SessionId {SessionId} from {DeviceId}", feedback.SessionId, feedback.DeviceId);
+                return new StatusCodeResult((int)HttpStatusCode.BadRequest);
+            }
+
+            var session = sessionRow.GetSession();
+            var presenterNames = string.Join(", ", session.PresenterIds.Select(pId => allPresenters.Single(p => p.Id == pId)).Select(x => x.Name));
+
+            // Save feedback
+            log.LogInformation("Successfully received feedback for {SessionId} from DeviceId {DeviceId}; persisting...", feedback.SessionId, feedback.DeviceId);
+            var feedbackToPersist = new SessionFeedbackEntity(
+                session.Id.ToString(),
+                conference.ConferenceInstance,
+                feedback.Name,
+                feedback.Rating.ToString(),
+                feedback.Likes,
+                feedback.ImprovementIdeas,
+                $"{session.Title} - {presenterNames}");
+
+            var (_, sessionFeedbackRepo) = await feedbackConfig.GetRepositoryAsync();
+            await sessionFeedbackRepo.CreateAsync(feedbackToPersist);
+
+            return new StatusCodeResult((int)HttpStatusCode.NoContent);
+        }
+    }
+
+    public class SessionFeedbackRequest
+    {
+        // not used for now
+        public Guid DeviceId { get; set; }
```

Comment:
Add it to the entity so it's recorded. It'll be null for the microsoft forms way obv

Reply:
All good :) added